src/gui/windows/predictor/predictor_info_dock.cpp

```diff
@@ -0,0 +1,334 @@
+        value_event_predict_index_bht->setStyleSheet(color_stylesheet);
+    }
+    value_event_predict_result->setStyleSheet(color_stylesheet);
+}
+
+void DockPredictorInfo::set_update_widget_color(QString color_stylesheet) {
+    value_event_update_instruction->setStyleSheet(color_stylesheet);
+    value_event_update_address->setStyleSheet(color_stylesheet);
+    value_event_update_index_btb->setStyleSheet(color_stylesheet);
+    if (is_predictor_dynamic) {
+        value_event_update_index_bht->setStyleSheet(color_stylesheet);
+    }
+    value_event_update_result->setStyleSheet(color_stylesheet);
+}
+
+void DockPredictorInfo::setup(
+    const machine::BranchPredictor *branch_predictor,
+    const machine::Core *core) {
+
+    clear();
+
+    number_of_bhr_bits = branch_predictor->get_number_of_bhr_bits();
+    initial_state = branch_predictor->get_initial_state();
+    const machine::PredictorType predictor_type { branch_predictor->get_predictor_type() };
+    is_predictor_dynamic = predictor_type == machine::PredictorType::SMITH_1_BIT
+                            || predictor_type == machine::PredictorType::SMITH_2_BIT
+                            || predictor_type  == machine::PredictorType::SMITH_2_BIT_HYSTERESIS;
+    is_predictor_enabled = branch_predictor->get_enabled();
+
+    if (is_predictor_enabled) {
+        connect(
+            branch_predictor, &machine::BranchPredictor::total_stats_updated,
+            this, &DockPredictorInfo::update_stats);
+        connect(
+            branch_predictor, &machine::BranchPredictor::prediction_done,
+            this, &DockPredictorInfo::show_new_prediction);
+        connect(
+            core, &machine::Core::step_started,
+            this, &DockPredictorInfo::reset_colors);
+        connect(
+            branch_predictor, &machine::BranchPredictor::update_done,
+            this, &DockPredictorInfo::show_new_update);
+
+        if (is_predictor_dynamic) {
+            
+            connect(
+                branch_predictor, &machine::BranchPredictor::bhr_updated,
+                this, &DockPredictorInfo::update_bhr);
+        }
+    }
+
+    if (is_predictor_enabled) {
+        content->setDisabled(false);
+    } else {
+        content->setDisabled(true);
+    }
+
+    if (is_predictor_dynamic) {
+        label_event_predict_index_bht->setEnabled(true);
+        value_event_predict_index_bht->setEnabled(true);
+        label_event_update_index_bht->setEnabled(true);
+        value_event_update_index_bht->setEnabled(true);
+        label_bhr->setEnabled(true);
+        value_bhr->setEnabled(true);
+    } else {
+        label_event_predict_index_bht->setEnabled(false);
+        value_event_predict_index_bht->setEnabled(false);
+        label_event_update_index_bht->setEnabled(false);
+        value_event_update_index_bht->setEnabled(false);
+        label_bhr->setEnabled(false);
+        value_bhr->setEnabled(false);
+    }
+
+    if (number_of_bhr_bits == 0) {
```

Comment:
I would consider making this `&& is_predictor_dynamic` so that we set the value at just one place. Also those two branches could be merged and just set the value to be `is_predictor_dynamic`, but I am not 100% convinced here.

Reply:
Fixed